functions.php

```diff
@@ -143,6 +144,53 @@ protected function hooks() {
 		) );
 	}
 
+	/**
+	 * Sets as featured image of the post the first image found attached in the post's content (if any).
+	 *
+	 * @param int     $post_id The ID of the current Post.
+	 * @param WP_Post $post The current Post.
+	 * @param bool    $update Whether this is an existing post being updated or not.
+	 */
+	public function set_featured_image( $post_id, $post, $update ) {
+		global $wpdb;
+
+		// Ignore autosave.
+		if ( defined( 'DOING_AUTOSAVE' ) && DOING_AUTOSAVE ) {
+			return;
+		}
+		// Check user's capabilities.
+		if ( ! current_user_can( 'edit_post', $post_id ) ) {
+			return;
+		}
+
+		// Check if user has set the featured image manually or if he has removed it.
+		if ( isset( $_POST['user_set_featured_image'] ) ) {
+			update_post_meta( $post_id, 'user_set_featured_image', true );
+		} elseif ( isset( $_POST['user_removed_featured_image'] ) ) {
+			update_post_meta( $post_id, 'user_set_featured_image', false );
+		}
+		$user_featured_image = get_post_meta( $post_id, 'user_set_featured_image', true );
+
+		// Apply this behavior to Posts only.
+		if ( 'post' === $post->post_type && ! $user_featured_image ) {
+
+			// Find all matches of <img> html tags within the post's content and get the url inside the src attribute.
+			preg_match_all( '/<img.+src=[\'"]([^\'"]+)[\'"].*>/i', $post->post_content, $matches );
+			$first_img_url = $matches[1][0];
```

Comment:
If no image found, it will throw a notice level error.

Reply:
Yes, saw that, will fix it.